dbt/tests/data_tests/generic_tests/schema.yml

```diff
@@ -68,3 +68,26 @@ macros:
   - name: test_expect_valid_hour_of_year
     description: >
       Check whether ``hour_of_year`` is consistent with ``datetime_utc``. Designed for use on ``out_vcerare__hourly_available_capacity_factor``. Fails when the hour computed from the datetime does not match.
+  - name: test_expect_complete_valid_plantparts_ownership
+    description: >
+      Check whether ``fraction_owned`` columns sum too high (for any individual plant
+      part) or too low (for too many plant parts). Designed for use on
+      ```out_eia__yearly_plant_parts``. Fails when any owned plant part with nonzero
+      capacity has sum(fraction_owned) more than 1e-6 away from 1, or when more than
+      60 plant parts have capacity 0 and fraction_owned 0.
```

Comment:
A more clear name might be `expect_valid_plantparts_ownership_fraction`

...this looks similar to the other `test_expect_sum_to_target_discrepancy_rate_less_than` test, but I'm assuming that they have to be treated separately due to the way we handle plants with capacity 0? 

It does seem similar though. Probably OOS for now, but I bet we could design a generic "make sure these columns sum up to approximately this given this threshold/min/max value and also exclude these rows" function.

Reply:
There are two components of the test:

- Expect valid ownership records: Fail when a plant part has capacity but the sum of fraction_owned is not equal to 1
- Expect complete ownership records: Fail when too many plant parts have placeholder (zero) values for capacity and fraction_owned

So maybe `expect_complete_and_valid_plantparts_ownership`?

Reply:
(eventually yeah refactor to test the sum and the completeness separately)

Reply:
I think the word `valid` implies completeness. I also think we need the word "fraction" in the title.

It's a little confusing that this test also looks at capacity. I would be in favor of separating the two test components now if it isn't too much work.

Reply:
Will change the name and otherwise leave the test as is.